packages/@webex/plugin-cc/test/unit/spec/services/WebCallingService.ts

```diff
@@ -0,0 +1,137 @@
+import 'jsdom-global/register';
+import WebCallingService from '../../../../src/services/WebCallingService';
+import {createClient, ICallingClient, ILine, LINE_EVENTS, ICall} from '@webex/calling';
+import {WebexSDK} from '../../../../src/types';
+
+jest.mock('@webex/calling');
+
+describe('WebCallingService', () => {
+  let webex: WebexSDK;
+  let callingClient: ICallingClient;
+  let line: ILine;
+  let webRTCCalling: WebCallingService;
+
+  beforeEach(() => {
+    webex = {
+      logger: {
+        log: jest.fn(),
+        error: jest.fn(),
+      },
+    } as unknown as WebexSDK;
+
+    callingClient = {
+      getLines: jest.fn().mockReturnValue({
+        line1: {
+          on: jest.fn(),
+          register: jest.fn(),
+          deregister: jest.fn(),
+        },
+      }),
+    } as unknown as ICallingClient;
+
+    (createClient as jest.Mock).mockResolvedValue(callingClient);
+
+    webRTCCalling = new WebCallingService(webex, {});
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.useRealTimers();
+  });
+
+  describe('registerWebCallingLine', () => {
+    it('should register the web calling line successfully', async () => {
+      line = callingClient.getLines().line1 as ILine;
+      const deviceInfo = {
+        mobiusDeviceId: 'device123',
+        status: 'registered',
+        setError: jest.fn(),
+        getError: jest.fn(),
+        type: 'line',
+        id: 'line1',
+      };
+
+      const registeredHandler = jest.fn();
+      line.on.mockImplementation((event, handler) => {
```

Comment:
We need not to mock the line.on. We can spy on line.on and invoke the callback using mock.calls on the spy object